src/java/htsjdk/variant/variantcontext/GenotypeLikelihoods.java

```diff
@@ -72,6 +72,13 @@
     */
     private final static GenotypeLikelihoodsAllelePair[] PLIndexToAlleleIndex = calculatePLcache(MAX_ALT_ALLELES_THAT_CAN_BE_GENOTYPED);
 
+
+    /*
+    * a cache of the PL index to the anyploid alleles it represents over all possible numbers of alternate alleles
+    */
+    protected final static List<List<Integer>> plIndexToAlleleIndices = new ArrayList<List<Integer>>();
```

Comment:
well, if the developer needs more than one ploidy, what else should they
do? if they do it without needing, that's their fault. I would put a
comment in the initialization function that makes it clear that one should
only initialize ploidies that one is interested in...



On Sat, Feb 20, 2016 at 6:12 PM, Ron Levine <notifications@github.com>
wrote:

> In src/java/htsjdk/variant/variantcontext/GenotypeLikelihoods.java
> <https://github.com/samtools/htsjdk/pull/466#discussion_r53559062>:
>
> > @@ -72,6 +72,13 @@
> >      */
> >      private final static GenotypeLikelihoodsAllelePair[] PLIndexToAlleleIndex = calculatePLcache(MAX_ALT_ALLELES_THAT_CAN_BE_GENOTYPED);
> >
> > +
> > +    /*
> > +    * a cache of the PL index to the anyploid alleles it represents over all possible numbers of alternate alleles
> > +    */
> > +    protected final static List<List<Integer>> plIndexToAlleleIndices = new ArrayList<List<Integer>>();
>
> Yes, the ploidy would have to be saved. My logic is to only deal with one
> ploidy. With the map, the application developer could initialize for
> multiple ploidys, which would waste memory.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/samtools/htsjdk/pull/466/files#r53559062>.
>